sassdocs_helpers.rb

```diff
@@ -0,0 +1,110 @@
+require 'json'
+
+module SassdocsHelpers
+    def format_sassdoc_data(data)
+        raise "No data in data/sassdocs.json, run `npm install` to generate." unless data.respond_to?(:sassdoc)
+        data.sassdoc
+            .select { |doc|
+                # Remove private API entries
+                doc.access == 'public'
+            }
+            .select { |doc|
+                # Remove vendored files, for example SassMQ
+                !doc.file.path.start_with?('vendor')
+            }
+            .group_by {
+                # Group the docs by their 'group', for example 'Settings' or 'Helpers'
+                |doc| doc.group.first
+            }
+    end
+    def mixin_trailing_code(code)
+        # Some mixins can have content passed inside.
+        # For example the `govuk-if-ie8` function:
+        # @include govuk-if-ie8 {
+        #  //..
+        # }
+        acceptsContent = code.include?('@content')
+        if acceptsContent
+            " {\n  //..\n}"
+        else
+            ";"
+        end
+    end
+    def inline_parameters(parameters)
```

Comment:
Do you think it'd make sense to have one helper that's responsible for building the entire example, including any parameters and inline code? It could still call out to private methods, so that we can keep the methods nice and short.

It feels a little strange at the minute as the logic is kind of split, with some of it handled in the helper but some of it being handled in the view.

It'd also mean that we'd have better test coverage, as the logic that's currently composing these in the view can't be tested.

Happy to leave this as is if you'd prefer, and we can refactor it later.

Reply:
Will investigate this.